src/torchmetrics/functional/image/ssim.py

```diff
@@ -150,7 +150,9 @@ def _ssim_compute(
             kernel = _gaussian_kernel_2d(channel, gauss_kernel_size, sigma, dtype, device)
 
     if not gaussian_kernel:
-        kernel = torch.ones((1, 1, *kernel_size)) / torch.prod(Tensor(kernel_size))
+        kernel = torch.ones((channel, 1, *kernel_size), dtype=dtype, device=device) / torch.prod(
+            torch.tensor(kernel_size, dtype=dtype, device=device)
+        )
```

Comment:
```suggestion
        kernel = torch.ones((channel, 1, *kernel_size), dtype=dtype, device=device) / math.prod(kernel_size)
```

shouldn't be necessary to actually allocate a larger gpu-tensor here :)

Reply:
Hi, @justusschock - Thanks for the suggestion. Let's say `kernel_size` is (100, 100), and the following will output: `tensor(10000)` which I think is fine in terms of allocation. Is there anything I'm missing? Please feel free to correct me.

```bash
kernel_size = (100, 100)
dtype = torch.int64
device = "cuda"
torch.prod(torch.tensor(kernel_size, dtype=dtype, device="cuda"))
```

Reply:
lets use torch :)

Reply:
@Borda - Done! (committed)

@justusschock - If you feel otherwise, please let me know. Happy to know your thoughts and consider them. :) Thanks!